lib/client.js

```diff
@@ -118,11 +117,11 @@ Client.prototype.request = function(method, url, data, callback) {
 
   // Resolve data as promised
   const promises = this.promisifyData(data);
+  const that = this
   if (promises.length) {
     return Promise.all(promises)
-      .bind(this)
       .then(() => {
-        this.request(method, url, data, callback);
+        that.request(method, url, data, callback);
```

Comment:
@PatrickHeneise can you say, why was this necessary?

Reply:
This replaces Bluebird `.bind(this)` to have "this" context available in the function (Arrow functions do [not have their own bindings to this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions).